hrms/payroll/doctype/gratuity/gratuity.py

```diff
@@ -159,7 +159,7 @@ def get_work_experience(self) -> float:
 		if rule.method == "Round off Work Experience":
 			work_experience = round(work_experience)
 		else:
-			work_experience = floor(work_experience)
+			work_experience = flt(work_experience, precision=1)
```

Comment:
```suggestion
			work_experience = flt(work_experience, precision=self.precision("current_work_experience"))
```
Instead of using a hardcoded precision let's set field level precision. You can set 1 as the default precision in core for this field. So when people want to have more precision, they can just customize field level precision and this calculation will get adjusted